### PR TITLE
feat: support os color sync

### PR DESCRIPTION
**Pull request by miurahr**
This enhance OmegaT  to detect personal preference of colors on OS settings and allow following color preference when selecting theme.
This extend OmegaT appearance preference to select two defaults: one is light default, and other is dark default.

## Pull request type

- Feature enhancement -> [enhancement]

## Which ticket is resolved?
 
- GUI: automatic colour scheme adaptive change on OS color preferece
  * https://sourceforge.net/p/omegat/feature-requests/1675/

## What does this PR change?

- Detect OS color preference
- Sync with OS color
- Config light and dark defaults
- Use JNA to access OS registry or macOS foundation library to detect OS's personal color preference
- Add FlatLaf dependency for dark theme implementation
- Require Java 11

## Other information

TODO

- [x] Extend Settings->Configuration->Appearance to allow select multiple theme
- [x] Switch behavior with a option  "sync with OS"

### CODE REVIEW

**Comment by miurahr** [comment]
Now on Windows, when default configuration,  it can realize automatic dark/light change when startup.

1. start OmegaT
2. go to Windows menu->preference->personalize->color-> select "dark"
3. restart OmegaT
4. you will see OmegaT start with dark theme 
5. change Windows menu->preference -> personalize -> color -> select "light"
6. restart OmegaT
7. you will see OmegaT start with standard theme

**Comment by miurahr** [comment]
A default of light theme is OmegaT standard theme, and a default of dark theme is a "Flat Dark Theme" that introduced in the PR. A "Sync with OS color" feature is enabled when OmegaT preferences->appearance->"Sync with OS color" radio button is on.

**Comment by miurahr** [comment]
Recreate two clean commits to merge for help maintain commit history.

File: build.gradle [hunk 1]
```diff
@@ -256,6 +259,11 @@ dependencies {
             }
         }
 
+        // Platform integration with Windows and macOS
+        implementation 'net.java.dev.jna:jna-platform:5.9.0'
```
Comment (miurahr) [inline build.gradle:263]: `jna-platform` module has  a native libraries for macOS and windows.
It may be necessary to sign "*.dylib" for macOS...

**Review by miurahr — COMMENTED** [review]
I think we need more investigation for code signing requirements.
It can be handled in another PR.

**Comment by miurahr** [comment]
I use multiple ComboBox for selection of themes.  One for default/light theme and another is dark one.

I borrowed dark/light detector code from JetBrains Intellij-commuity project, Apahche-2.0. I put its name in copyright header.

File: build.gradle (hunk 1 above)
Comment (miurahr) [reply build.gradle:260]: It is signed and no need to do anything.

**Comment by miurahr** [comment]
There are UI changes to be reviewed

- Options>Preference>Appearance
- Behavior change on Windows 10/11
- New theme plugins that can recognize dark/light OS color preference on windows and mac, and these themes also fix the color issue on windows 10/11 (bugs#1196)

**Comment by miurahr** [comment]
This PR is ready for review and I'd like to propose to merge into 6.1 development cycle.

**Comment by miurahr** [comment]
JNA 5.13.0 supports
- aix-ppc
- aix-ppc64
- darwin-aarch64
- darwin-x86-64
- freebsd-x86
- freebsd-x86-64
- linux-aarch64
- linux-arm
- linux-armel
- linux-loongaarch64
- linux-mips64el
- linux-ppc
- linux-ppc64le
- linux-riscv64
- linux-s390x
- linux-x86
- linux-x86-64
- openbsd-x86
- openbsd-x86-64
- sunos-sparc
- sunos-sparcv9
- sunos-x86
- sunos-x86-64
- win32
- win32-aarch64
- win32-x86
- win32-x86-64

**Comment by miurahr** [comment]
Flatlaf library bundles
- windows x86-64
- windows x86
- linux x86-64
native libraries

**Comment by miurahr** [comment]
@Kazephil  could you review preference UI changes?

**Review by Kazephil — COMMENTED** [review]
I just edited the `changes.txt`and `Bundle.properties` file to make proposed wording changes and fix a minor typo or two.

The switch between the light and dark theme looks good. Unfortunately, it doesn't work on Linux (or at least not on KDE in Arch Linux). The "sync OS preference" option is greyed out.

Selecting the radio button to use the dark theme and reloading OmegaT worked nicely. However, OmegaT did not tell me it had to be reloaded (possibly because I didn't have a project open). I haven't tried this on Windows yet, so I don't know if it's just a side effect of the syncing feature not working on Linux, but we should make sure that OmegaT alerts the user to the need to restart the application for the theme change to take effect.

There was one more odd behaviour when I tried this branch:

After I start OmegaT and open the Preferences dialog, clicking on _Appearance_ does nothing. The options don't show in the pane on the right until I click on one of the other options, and then click an _Appearance_ one more time.

Every other option shows the relevant information in the pane on the right immediately.

**Comment by miurahr** [comment]
> The switch between the light and dark theme looks good. Unfortunately, it doesn't work on Linux (or at least not on KDE in Arch Linux). The "sync OS preference" option is greyed out.

Yes, It is intended behavior for Linux.  Standard L&F on Linux is GTK L&F and it is aware of dark/light preference of Gnome., and library I use does not support Linux. I do not know how it is with KDE.

>Selecting the radio button to use the dark theme and reloading OmegaT worked nicely. However, OmegaT did not tell me it had to be reloaded (possibly because I didn't have a project open). 

> we should make sure that OmegaT alerts the user to the need to restart the application for the theme change to take effect.

Thanks for the feedback that is work nicely. and it is a point I need to adjust to tell user to be reload.

> After I start OmegaT and open the Preferences dialog, clicking on Appearance does nothing. The options don't show in the pane on the right until I click on one of the other options, and then click an Appearance one more time.

This is bug. I would like to investigate it later.

**Comment by miurahr** [comment]
I've pushed the change

- avoid NPE when calling first, that is happened because I set `actionListener` before `initFromPrefs` that fire the listener action, which lead a NPE. 
- Same lead a bug when open the preference dialog after restart, it show  a wrong "restart required" message.

File: src/org/omegat/util/gui/laf/WindowsDarkThemeDetector.java [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+import com.sun.jna.platform.win32.WinReg;
+
+import org.omegat.util.JnaLoader;
+
+/**
+ * Copyright 2000-2021 JetBrains s.r.o. Use of this source code is governed by
+ * the Apache 2.0 license.
+ */
+public class WindowsDarkThemeDetector extends SystemDarkThemeDetector {
+    private final static String REGISTRY_PATH = "Software\\Microsoft\\Windows\\CurrentVersion\\Themes"
```
Comment (miurahr) [inline src/org/omegat/util/gui/laf/WindowsDarkThemeDetector.java:40]:
```suggestion
    private static final String REGISTRY_PATH = "Software\\Microsoft\\Windows\\CurrentVersion\\Themes"
```

File: src/org/omegat/util/gui/laf/WindowsDarkThemeDetector.java [hunk 3]
```diff
@@ -0,0 +1,62 @@
...
+            + "\\Personalize";
+    private final static String REGISTRY_VALUE = "AppsUseLightTheme";
```
Comment (miurahr) [inline src/org/omegat/util/gui/laf/WindowsDarkThemeDetector.java:42]:
```suggestion
    private static final String REGISTRY_VALUE = "AppsUseLightTheme";
```

File: gradle/libs.versions.toml [hunk 4]
```diff
@@ -58,6 +58,9 @@ mnemonics = "1.1"
 jmyspell = "1.0.0-beta-2"
 hunspell = "2.1.1"
 xjc = "2.3.4"
+tipoftheday = "0.4.1"
```
Comment (miurahr) [inline gradle/libs.versions.toml:61]:
```suggestion
```

**Comment by miurahr** [comment]
Resolved merge errors.
- adjust checkstyle errors/warnings
- merge version catalogs

any other brockers for merge?

**Pull request merged**